system/lib/pthread/library_pthread.c

```diff
@@ -825,17 +825,26 @@ EMSCRIPTEN_KEEPALIVE void* _emscripten_main_thread_futex;
 
 void __emscripten_init_main_thread_js(void* tb);
 
+static void *dummy_tsd[1] = { 0 };
+weak_alias(dummy_tsd, __pthread_tsd_main);
+
 // See system/lib/README.md for static constructor ordering.
 __attribute__((constructor(48)))
 void __emscripten_init_main_thread(void) {
   __emscripten_init_main_thread_js(&__main_pthread);
+  //printf("__emscripten_init_main_thread %p\n", __main_pthread);
+
   // The pthread struct has a field that points to itself - this is used as
   // a magic ID to detect whether the pthread_t structure is 'alive'.
   __main_pthread.self = &__main_pthread;
+  __main_pthread.detach_state = DT_JOINABLE;
```

Comment:
This should not be needed, as default thread state is already `DT_JOINABLE`.
https://github.com/emscripten-core/emscripten/blob/a47616119653aa89542dfac531722520fb16f0af/src/library_pthread.js#L804
(unless, I'm overlooking something)

Reply:
That code in `library_pthread.js` is only run for pthread_create, not when the main/primordial thread is initialized.    This code here is *only* for main/primordial thread.

Reply:
Ah, I see. Looks like it was not previously initialized in musl's `__init_tp` (where we mirror this from), but now it is.
https://github.com/emscripten-core/emscripten/blob/3cc9f7ab560d634f94461848116d7b69c195cc58/system/lib/libc/musl/src/env/__init_tls.c#L21

Reply:
> Ah, I see. Looks like it was not previously initialized in musl's `__init_tp` (where we mirror this from), but now it is.
> 
> https://github.com/emscripten-core/emscripten/blob/3cc9f7ab560d634f94461848116d7b69c195cc58/system/lib/libc/musl/src/env/__init_tls.c#L21

I was not aware of `__init_tp`.. I guess we could switch to that at some point in the future rather than duplicating it (in two places).